weio/bladed_out_file.py

```diff
@@ -0,0 +1,288 @@
+        return ChannelName, Precision, SectionList,NoOfSections,category,DataLength,SensorNumber, NDIMENS, ChannelUnit                
+                
+##  ----- a function to organize data -------------------------------- 
+    def OrgData(self,DataLength,SectionList,ChannelName,data_tmp,NDIMENS,ChannelUnit):
+            
+        # since some of the matrices are 3 dimensional, we want to make all 
+        # to 2d matrix, so I am organizing them here:
+            if NDIMENS == 3:
+                
+                data_T = np.transpose(data_tmp)
+                SName = []
+                SUnit = []
+                self.DataOut = np.ones( (DataLength,len(SectionList)*len(ChannelName)) ) 
+                
+                col_vec = -1
+                for ss in range(len(SectionList)):
+                    for cc in range(len(ChannelName)):                    
+                
+                        SName.append(str(SectionList[ss]) + 'm-' + ChannelName[cc])
+                        SUnit.append(str(SectionList[ss]) + 'm-' + ChannelUnit[cc])
+                        col_vec +=1
+                        self.DataOut[:,col_vec] = data_T[:,ss,cc]
+                        
+            elif NDIMENS == 2:
+                data_T = np.transpose(data_tmp)
+                SName = []
+                self.DataOut = np.ones( (DataLength,len(ChannelName)) ) 
+                col_vec = -1
+                
+                self.DataOut = data_T
+                SName = ChannelName
+                SUnit = ChannelUnit
+            
+            self.SensorName = SName
+            self.SensorUnit = SUnit
+
+
+
+
+# main function to read the data and call other functions:
+    def DataValue(self):
+        
+        # call "read" function to find all %, $ files
+        # self.read()
+        
+        ChannelName, Precision, SectionList, NoOfSections, category, DataLength, SensorNumber, NDIMENS, ChannelUnit  = self.SensorLib()
+        
+        # print(self.files_in)
+        ## Read only files that ends with $:
+        file_in = self.files_in.replace('%','$')
+        fid_2 = open(os.path.join(self.Folder_in,file_in), 'r')
+        
+        if Precision == 4:
+            tmp_2 = np.fromfile(fid_2, np.float32)
+        elif Precision == 8:
+            tmp_2 = np.fromfile(fid_2, np.float64)
+            
+            
+        if NDIMENS == 3:
+            tmp_3 = np.reshape(tmp_2,(SensorNumber, NoOfSections, DataLength), order='F')
+            data_tmp = tmp_3
+            # self.data_T = np.transpose(tmp_3)
+        
+           
+            
+        elif NDIMENS == 2:
+            tmp_3 = np.reshape(tmp_2,(SensorNumber, DataLength), order='F')
+            data_tmp = tmp_3
+        
+        self.OrgData(DataLength,SectionList,ChannelName,data_tmp,NDIMENS,ChannelUnit)
+    
+    
+  
+    
+  
+# %% find all %, $ files:
+    
+    def read(self):
+        file_name_0 = self.file_name
```

Comment:
I don't understand the Bladed format very well yet, But I wonder if a small modification of your function could allow the read function, to also be given other files such as the $i and %i files. If the $PJ needs to be read no matter what, the function can simply look if a $PJ file exist, and then continue "as usual". A pyDatView user can then drag and drop any bladed file, a single output file, or all of them. 

But that's a bonus, for future work! Like I said, I don't understand the format very well, so maybe that doesn't make sense.

Reply:
$PJ file does not to be read ( does not need to be exist). Imagine them as a htc file. So I think if user just drag and drop one $ or % file, pyDatView should find other files similar to that name but with different number.
For example, if I drag and drop _Bladed_out_ascii.%04_, pyDatView  should pick up _Bladed_out_ascii.%41_ as well.
That's why I wrote `read`  function in that way (but feel free to change it):
```
        for Folder, sub_folder, files in os.walk(pth):
            for i in range(len(files)): 
                if searchName in files[i]:
```
I have changed the code, and user could add any file with any extension ($PJ, or % or $).